CMakePresets.json

```diff
@@ -0,0 +1,22 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {"major": 3, "minor": 21, "patch": 0},
```

Comment:
If we are guaranteed to at least get this version anyway are we specifying this explicitly just to make that clearer to other developers working on the cmake build system?

Reply:
Yes.